app/v2/controllers/V2ArrivalsController.scala

```diff
@@ -190,4 +196,17 @@ class V2ArrivalsControllerImpl @Inject() (
           )
     }
 
+  def getArrivalMessage(arrivalId: MovementId, messageId: MessageId): Action[AnyContent] =
+    (authActionNewEnrolmentOnly andThen acceptHeaderActionProvider()).async {
+      implicit request =>
+        implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
+        (for {
+          messageSummary          <- arrivalsService.getArrivalMessage(request.eoriNumber, arrivalId, messageId).asPresentation
+          formattedMessageSummary <- responseFormatterService.formatMessageSummary(messageSummary, request.headers.get(HeaderNames.ACCEPT).get)
```

Comment:
This Formatter is created for Json wrapped xml and that is not asked for this user story. if you are using this formatter then in Arrivals routers you should use case pattern matching Some(VERSION_2_ACCEPT_HEADER_PATTERN()) =>. 
@drnaylor-oc  is it ok to implement formatted Message summary as part of this?

Reply:
Note that it won't work for non 007 arrival messages at the moment anyway, because they need to be added to the converter!

...which I need to make a ticket for.

For what it's worth though -- I don't think there is any harm in leaving it if it's done. No point wasting work if it's been done.

Reply:
The issue with calling formatMessageSummary now is that it converts the xml to json if the accept header is set to +json, which is currently is. Once we implement CTCP-1705, we will introduce a breaking change where json+xml will return the current implementation (message in xml), and +json will be modified to return the message in json.

Reply:
@prashanthmamidi In that case, as mentioned by Sandeep, you need to update the router to route to V2 based on the V2 regex pattern rather than solely by +json header.